src/lib/alloc.c

```diff
@@ -191,9 +191,41 @@ static void *rmalloc_sys(int zone, int caps, int core, size_t bytes)
 	return ptr;
 }
 
+static void *align_buffer_ptr(struct mm_heap *heap, uint32_t alignment,
+			      void *ptr)
+{
+	struct alignment_hdr *a_hdr;
+	void *aligned_ptr = ptr;
+	int i, mod_align;
+
+	/* Checking alignment for buffer heap only */
+	for (i = 0;  i < PLATFORM_HEAP_BUFFER;  i++) {
+		if (heap == &memmap.buffer[i]) {
+
+			/* add hdr to a ptr */
+			ptr = ptr + sizeof(*a_hdr);
+			/* calculate alignment shift for headered ptr */
+			if (alignment <= 1)
```

Comment:
We should only allow alignment to be power of 2. With such restriction you don't need this check here.
`alignment && !(alignment & (alignment - 1))`

Reply:
added check for power of 2 as separate if since check for alignment 0 is needed when we need no alignment.